src/urllib3/connection.py

```diff
@@ -198,22 +258,41 @@ def putheader(self, header, *values):
                 )
             )
 
-    def request(self, method, url, body=None, headers=None):
+    def request(
+        self,
+        method: str,
+        url: str,
+        body: Optional[_DefaultType] = None,
+        headers: Optional[Mapping[str, str]] = None,
+        *,
```

Comment:
The `encode_chunked` parameter wasn't available before, I don't think we should add it.

Reply:
Yes, it was added to `http.client` in `3.6`. If it is not added then `request` method violates LSP because of incompatible signatures. I can't think of a workaround that wouldn't require `# type: ignore` or a refactor. Having two ways to do the same thing is also not ideal. What about refactoring the `request` method to do all the work including sending chunked requests and then making `request_chunked` a convenience method for compatibility that would call `request`?

Reply:
I think this is a case where we want to violate LSP as our API in the future is different from `http.client.HTTPConnection`'s API and our subclassing for now is incidental (see BaseHTTPConnection). I'm comfortable having a `type: ignore` along with a comment about us intentionally violating LSP

Reply:
Makes sense. Will add comments and type ignores.

By the way, do I understand it correctly that in the future when `urllib3.HTTPConnection` will inherit from `BaseHTTPConnection` it would have to contain `http.client.HTTPConnection` (namely, the relationship will be composition instead of inheritance)?

Reply:
@franekmagiera Things will remain how they are for 2.x and then in 3.x we'll be using a different HTTP implementation `h2` or `h11` instead of only `http.client.HTTPConnection`. `BaseHTTPConnection` can be used as a parent class for the new HTTP connections.

Reply:
Added the comment and type ignore for the `request` method.